### DISCUSSION TITLE
GFW或者ISP是否可能用机器学习等手段检测出隐藏在Xray里面的Tor

### DISCUSSION THREAD

**Discussion by i39v** [XTLS/Xray-core#2351] [Q&A]
~~我不是相关专业，这是我猜的，可能理论上可行~~
~~Xray也是举个例子，也可以换成Naive或者别的翻墙软件~~
我的Xray用的VLESS-XTLS-uTLS-REALITY配置

就像去年大规模封TLS代理那样，通过大量机器学习找出Xray + Tor的一些特别的流量特征来检测出躲在Xray后面的Tor？
另外，Whonix的维基也[提到](https://www.whonix.org/wiki/Hide_Tor_from_your_Internet_Service_Provider)隐藏Tor很难，不过这里面没有提到类似Xray这样伪装成正常上网流量来突破中国和伊朗严格网络封锁的软件

我个人觉得一个可能的缓解办法是用Tor的Obfs4网桥来代替Tor的入口节点，即Xray--Obfs4网桥--Tor中间节点--Tor出口节点，~~这也是猜的~~

**Comment by chika0801** [comment]
你的意思是不是这样理解，GFW自己买些VPS，按网上教程搭如Xray协议。它的工作人员自己用，自己分析产生的流量，来找特征？

**Reply by i39v** [reply]
~~GFW他们怎么找特征我就不清楚了，我只是猜测他们可以通过找特征的方法检测出Tor~~

**Comment by RPRX** [comment]
没研究过 Tor 协议细节，时序特征明显的话当然能揪出来

1. https://github.com/v2ray/v2ray-core/issues/2636 查看 Flow 那段
2. https://github.com/XTLS/Xray-core/discussions/2281#discussioncomment-6353098 `mux-controller` 是一个强大的流量整形器，它可以几乎完全掩盖内层时序特征
3. https://t.me/projectXtls/145 先期待一下 Xray-core v1.8.5 的 Vision Seed

**Reply by i39v** [reply]
感谢RPRX解答。
这么看来，假设Tor时序特征明显的话，在新版出之前还是继续开着Obfs4吧；另外好像可以用[Vanguards](https://www.whonix.org/wiki/Vanguards)给Tor再加一跳，可能会让Tor的特征出现变化，~~这也是猜的~~

**Comment by FranzKafkaYu** [comment]
> 3\. https://t.me/projectXtls/145 先期待一下 Xray-core v1.8.5 的 Vision Seed

R主席可以预告一下1.8.4与1.8.5的新feature么

**Reply by RPRX** [reply]
只有 1.8.5，就是 Vision Seed，合并一些 PR，其它一些小修改

**Comment by computerscot** [comment]
Tor obfs4 bridges have a rarely used feature for obfuscating packet lengths:

* `iat-mode=0` (no obfuscation)
* `iat-mode=1` (split data into fixed-length packets)
* `iat-mode=2` (split data into variable-length packets)

I believe the obfs4 bridge server needs to be specially set up to allow this, by including a configuration option:

```
ServerTransportOptions obfs4 iat-mode=2
```

The obfs4 bridge client can specify `iat-mode=2` in the bridge line.

I don't know if anyone has ever tried this.

Another option that probably no one has ever tried is to pass Xray traffic through an obfs4 forward proxy:

https://github.com/XTLS/Xray-core/discussions/1304

If anyone has the time to experiment, you could experiment with the possibilities here.
